examples/tfengine/app.hcl

```diff
@@ -20,46 +20,267 @@ template "app" {
+              {
+                name     = "example-instance-subnet"
+                ip_range = "10.3.0.0/16"
+              }
             ]
+            cloud_sql_private_service_access = {} # Enable SQL private service access.
+          }]
+          bastion_hosts = [{
+            name           = "bastion-vm"
+            network        = "$${module.example_network.network.network.self_link}"
+            subnet         = "$${module.example_network.subnets[\"us-central1/example-bastion-subnet\"].self_link}"
+            image_family   = "ubuntu-2004-lts"
+            image_project  = "ubuntu-os-cloud"
+            members        = ["group:bastion-accessors@example.com"]
+            startup_script = <<EOF
+sudo apt-get -y update
+sudo apt-get -y install mysql-client
+sudo wget https://dl.google.com/cloudsql/cloud_sql_proxy.linux.amd64 -O /usr/local/bin/cloud_sql_proxy
+sudo chmod +x /usr/local/bin/cloud_sql_proxy
+EOF
+          }]
+          compute_routers = [{
+            name    = "example-router"
+            network = "$${module.example_network.network.network.self_link}"
+            nats = [{
+              name                               = "example-nat"
+              source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+              subnetworks = [{
+                name                     = "$${module.example_network.subnets[\"us-central1/example-bastion-subnet\"].self_link}"
+                source_ip_ranges_to_nat  = ["PRIMARY_IP_RANGE"]
+                secondary_ip_range_names = []
+              }]
+            }]
           }]
         }
       },
       {
         name = "project_data",
+        resources = {
+          project = {
+            name_suffix = "data"
+            shared_vpc_attachment = {
+              host_project_suffix = "networks"
+            }
+          }
+          bigquery_datasets = [{
+            # Override Terraform resource name as it cannot start with a number.
+            resource_name               = "one_billion_ms_example_dataset"
+            dataset_id                  = "1billion_ms_example_dataset"
+            default_table_expiration_ms = 1e+9
+            labels = {
+              type = "phi"
+            }
+            access = [
+              {
+                role          = "roles/bigquery.dataOwner"
+                special_group = "projectOwners"
+              },
+              {
+                role           = "roles/bigquery.dataViewer"
+                group_by_email = "example-readers@example.com"
+              },
+            ]
+          }]
+          cloud_sql_instances = [{
+            name               = "example-mysql-instance"
+            type               = "mysql"
+            network_project_id = "example-prod-networks"
+            network            = "example-network"
+            labels = {
+              type = "no-phi"
+            }
+            # TODO(user): Uncomment and re-run the engine after deploying secrets.
```

Comment:
There is another pattern which I used in the recent solution, I'll send you the link internally.